nri-prelude/src/Task.hs

```diff
@@ -215,13 +215,11 @@ concurrently taskA taskB =
 
 -- | Given a task and a callback, execute the task in a greenthread
 -- and sends its result to callback.
--- Always returns @OK ()@
-background :: forall x a . Task x a -> (Result x a -> Task x a) -> Task x ()
+background :: Task x a -> (Result x a -> Task y b) -> Task z ()
```

Comment:
this is reasonable.  we have no desire to propagate the error back up since it's async ([there are ways to do this](https://www.fpcomplete.com/blog/2018/04/async-exception-handling-haskell/) but we probably don't need to worry about those)

Reply:
would could, in principle, change `z` to `Never`.  sending a task to the background should never fail, though it makes the api slightly inconvenient to work with